docs/memory-safety-continuum.md

```diff
@@ -0,0 +1,99 @@
+* Using a memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code
+* Using a memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code AND automated tooling to check for memory safety in third party code (dependencies)
+
+### Using a non-memory safe by default language without developer best practies or automated tooling
+
+Using raw C or C++ (or old versions of C and C++ language and compiler)
+
+### Using a non-memory safe by default language with developer best practices, but no automated tooling
+
+Examples:
+
+* [Using attributes such as `cleanup` and classes when writing C](https://lwn.net/Articles/934679/) (and depending on developers to manually check this)
+* Following the [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines) when writing C++
+* Using the [C++ Compiler Hardening Guide](https://github.com/ossf/wg-best-practices-os-developers/tree/main/docs/Compiler-Hardening-Guides) when compiling C++ code
+
+### Using a non-memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code
+
+TO DO
+
+### Using a non-memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code AND automated tooling to check for memory safety in third party code (dependencies)
+
+TO DO
+
+### Using a memory safe by default language without developer best practices or automated tooling
+
+Examples:
+
+* Using unsafe blocks in Rust [without assessing the entire module in which the unsafe code appears](https://github.com/ossf/Memory-Safety/issues/15#issuecomment-1847939439)
+* Using the [no_mangle](https://github.com/rust-lang/rust/issues/28179) attribute in Rust
+* Using compiler options which turn off some or all of the compiler's memory safety checks
+
+### Using a memory safe by default language with developer best practices, but no automated tooling
+
+Examples:
+
+* Following the [Rustnomicon](https://doc.rust-lang.org/nomicon/intro.html) careful practices when using unsafe blocks in Rust
+* Following best practices (LINK NEEDED) when using the Go [unsafe](https://pkg.go.dev/unsafe#pkg-overview) package
+* Following [Javascript Memory Management](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Memory_management) practices
+
+### Using a memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code
+
+Examples:
+
+* Using the [Go Data Race Detector](https://go.dev/doc/articles/race_detector)
+* Using other tools such as [govulncheck, fuzzing, and vet](https://go.dev/doc/security/best-practices) when writing Go code
+
+### Using a memory safe by default language with developer best practices and automated tooling to check for memory safety in first party code AND third party code
+
+TO DO
```

Comment:
Fuzzing components (including third-party code) that parse un-trusted data is a good way to find "low-hanging fruit" when auditing third party code is not feasible.
(One popular fuzzer is [AFL++](https://github.com/AFLplusplus/AFLplusplus).)

@nellshamrell , I think if I were to say much more than this, it would make sense to stick it in its own file in /docs... should I have a go and make a PR?